### PR TITLE
Fix #5101 - Update monthly reports for DistrictHeatingWater

### LINKED ISSUE #5101: DistrictHeating meter is renamed to DistrictHeatingWater
Issue overview
--------------
Translation of OSM to IDF is using old DistrictHeating meter name rather than the new DistrictHeatingWater meter name.  I'm not sure if DistrictCooling was also renamed as well.

## Current Behavior
Translation of OSM to IDF is generating:

```
Output:Table:Monthly,
  Building Energy Performance - District Heating, !- Name
  2,                                      !- Digits After Decimal
  InteriorLights:DistrictHeating,         !- Variable or Meter Name 1
  SumOrAverage,                           !- Aggregation Type for Variable or Meter 1
  ExteriorLights:DistrictHeating,         !- Variable or Meter Name 2
  SumOrAverage,                           !- Aggregation Type for Variable or Meter 2
  InteriorEquipment:DistrictHeating,      !- Variable or Meter Name 3
  SumOrAverage,                           !- Aggregation Type for Variable or Meter 3
  ExteriorEquipment:DistrictHeating,      !- Variable or Meter Name 4
  SumOrAverage,                           !- Aggregation Type for Variable or Meter 4
  Fans:DistrictHeating,                   !- Variable or Meter Name 5
  SumOrAverage,                           !- Aggregation Type for Variable or Meter 5
  Pumps:DistrictHeating,                  !- Variable or Meter Name 6
  SumOrAverage,                           !- Aggregation Type for Variable or Meter 6
  Heating:DistrictHeating,                !- Variable or Meter Name 7
  SumOrAverage,                           !- Aggregation Type for Variable or Meter 7
  Cooling:DistrictHeating,                !- Variable or Meter Name 8
  SumOrAverage,                           !- Aggregation Type for Variable or Meter 8
  HeatRejection:DistrictHeating,          !- Variable or Meter Name 9
  SumOrAverage,                           !- Aggregation Type for Variable or Meter 9
  Humidifier:DistrictHeating,             !- Variable or Meter Name 10
  SumOrAverage,                           !- Aggregation Type for Variable or Meter 10
  HeatRecovery:DistrictHeating,           !- Variable or Meter Name 11
  SumOrAverage,                           !- Aggregation Type for Variable or Meter 11
  WaterSystems:DistrictHeating,           !- Variable or Meter Name 12
  SumOrAverage,                           !- Aggregation Type for Variable or Meter 12
  Cogeneration:DistrictHeating,           !- Variable or Meter Name 13
  SumOrAverage;                           !- Aggregation Type for Variable or Meter 13

Output:Meter,
  DistrictHeating:Facility,               !- Key Name
  Timestep;  
```

## Expected Behavior
Translation of OSM to IDF should generate:

```
Output:Table:Monthly,
  Building Energy Performance - District Heating, !- Name
  2,                                      !- Digits After Decimal
  InteriorLights:DistrictHeatingWater,         !- Variable or Meter Name 1
  SumOrAverage,                           !- Aggregation Type for Variable or Meter 1
  ExteriorLights:DistrictHeatingWater,         !- Variable or Meter Name 2
  SumOrAverage,                           !- Aggregation Type for Variable or Meter 2
  InteriorEquipment:DistrictHeatingWater,      !- Variable or Meter Name 3
  SumOrAverage,                           !- Aggregation Type for Variable or Meter 3
  ExteriorEquipment:DistrictHeatingWater,      !- Variable or Meter Name 4
  SumOrAverage,                           !- Aggregation Type for Variable or Meter 4
  Fans:DistrictHeatingWater,                   !- Variable or Meter Name 5
  SumOrAverage,                           !- Aggregation Type for Variable or Meter 5
  Pumps:DistrictHeatingWater,                  !- Variable or Meter Name 6
  SumOrAverage,                           !- Aggregation Type for Variable or Meter 6
  Heating:DistrictHeatingWater,                !- Variable or Meter Name 7
  SumOrAverage,                           !- Aggregation Type for Variable or Meter 7
  Cooling:DistrictHeatingWater,                !- Variable or Meter Name 8
  SumOrAverage,                           !- Aggregation Type for Variable or Meter 8
  HeatRejection:DistrictHeatingWater,          !- Variable or Meter Name 9
  SumOrAverage,                           !- Aggregation Type for Variable or Meter 9
  Humidifier:DistrictHeatingWater,             !- Variable or Meter Name 10
  SumOrAverage,                           !- Aggregation Type for Variable or Meter 10
  HeatRecovery:DistrictHeatingWater,           !- Variable or Meter Name 11
  SumOrAverage,                           !- Aggregation Type for Variable or Meter 11
  WaterSystems:DistrictHeatingWater,           !- Variable or Meter Name 12
  SumOrAverage,                           !- Aggregation Type for Variable or Meter 12
  Cogeneration:DistrictHeatingWater,           !- Variable or Meter Name 13
  SumOrAverage;                           !- Aggregation Type for Variable or Meter 13

Output:Meter,
  DistrictHeatingWater:Facility,               !- Key Name
  Timestep;  
```

## Possible Solution
I'm not sure all the places the old name might be referenced, this might be one: https://github.com/NREL/OpenStudio/blob/develop/src/workflow/RunPreProcessMonthlyReports.hpp

### Environment

Some additional details about your environment for this issue (if relevant):
 - Platform (Operating system, version): Windows 11
 - Version of OpenStudio (if using an intermediate build, include SHA): 3.7.0

### Context
Cannot view DistrictHeating monthly end uses

May also need to update the OpenStudio Results measure to match

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

* **Companion PR**:  https://github.com/NREL/OpenStudio-workflow-gem/pull/149

- Fix #5101 
- Update monthly reports for DistrictHeatingWater

* Note for the name of the Output:Table:Monthly: AFAIK, this is OpenStudio::EndUseFuelType::valueDescription that is used by reporting measures so I made the change as well, perhaps some of them harcoded it to "District Heating", I'm not sure
* `Electricity Peak Demand` was missing Refrigeration EUT

To see the actual changes to the Output:Table:Monthly, just look at https://github.com/NREL/OpenStudio/pull/5106/commits/8d3ff631a5acd8ac1aa6c88f2234737dd4cae443?file-filters%5B%5D=.hpp&show-viewed-files=true

### Pull Request Author

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

File: developer/ruby/GeneratePreProcessMonthlyReports.rb
Comment (jmarrec) [inline developer/ruby/GeneratePreProcessMonthlyReports.rb:1]: This openstudio script will directly generate RunPreprocessMonthlyReports.hpp
and can also save the OpenStudio-workflow-gem/lib/openstudio/workflow/jobs/resources/monthly_report.idf directly (yes, we do have these in both the legacy ruby workflow and the C++ workflow.)

File: src/workflow/test/RunPreProcessMonthlyReports_GTest.cpp [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+      EXPECT_TRUE(eut) << "EndUseType should be present";
+      EXPECT_TRUE(ft) << "FuelType should be present";
+      EXPECT_FALSE(loc) << "InstallLocationType should not be present";
+
+      EXPECT_EQ(aggregrationType, eg.getString(1).get());
+    }
+  }
+
+  EXPECT_EQ(n_regular, n_demand);
+  EXPECT_EQ(4, n_regular);
```
Comment (jmarrec) [inline src/workflow/test/RunPreProcessMonthlyReports_GTest.cpp:137]: Annoying to write in C++, but this test will ensure that this doesn't happen again.

File: src/workflow/RunPreProcessMonthlyReports.hpp [hunk 2]
```diff
@@ -3,243 +3,259 @@
...
-      Humidifier:DistrictHeating,  !- Variable or Meter 10 Name
-      SumOrAverage,            !- Aggregation Type for Variable or Meter 10
-      HeatRecovery:DistrictHeating,!- Variable or Meter 11 Name
-      SumOrAverage,            !- Aggregation Type for Variable or Meter 11
-      WaterSystems:DistrictHeating,!- Variable or Meter 12 Name
-      SumOrAverage,            !- Aggregation Type for Variable or Meter 12
-      Cogeneration:DistrictHeating,!- Variable or Meter 13 Name
-      SumOrAverage;            !- Aggregation Type for Variable or Meter 13)idf",
+Output:Table:Monthly,
+  Building Energy Performance - District Heating Water, !- Name
```
Comment (jmarrec) [inline src/workflow/RunPreProcessMonthlyReports.hpp:67]: I renamed the table here. I think it's right, but would welcome comments from @DavidGoldwasser in particular

as indicated in the OP: AFAIK, this is `OpenStudio::EndUseFuelType::valueDescription` that is used by reporting measures so I made the change as well, perhaps some of them harcoded it to "District Heating", I'm not sure

File: src/workflow/RunPreProcessMonthlyReports.hpp [hunk 3]
```diff
@@ -3,243 +3,259 @@
...
+  2,                                      !- Digits After Decimal
+  InteriorLights:DistrictHeatingWater,    !- Variable or Meter Name 1
```
Comment (jmarrec) [inline src/workflow/RunPreProcessMonthlyReports.hpp:69]: Meter name is renamed to DistrictHeatingWater

File: src/workflow/RunPreProcessMonthlyReports.hpp [hunk 4]
```diff
@@ -3,243 +3,259 @@
...
+  Humidifier:Electricity,                 !- Variable or Meter Name 11
+  ValueWhenMaximumOrMinimum,              !- Aggregation Type for Variable or Meter 11
+  HeatRecovery:Electricity,               !- Variable or Meter Name 12
+  ValueWhenMaximumOrMinimum,              !- Aggregation Type for Variable or Meter 12
+  WaterSystems:Electricity,               !- Variable or Meter Name 13
+  ValueWhenMaximumOrMinimum,              !- Aggregation Type for Variable or Meter 13
+  Cogeneration:Electricity,               !- Variable or Meter Name 14
+  ValueWhenMaximumOrMinimum,              !- Aggregation Type for Variable or Meter 14
+  Refrigeration:Electricity,              !- Variable or Meter Name 15
+  ValueWhenMaximumOrMinimum;              !- Aggregation Type for Variable or Meter 15
```
Comment (jmarrec) [inline src/workflow/RunPreProcessMonthlyReports.hpp:160]: Refrigeration Was missing from Electricity Peak, I think that was a mistake, and it IS present in the Electricity report (energy)

File: src/workflow/RunPreProcessMonthlyReports.hpp [hunk 5]
```diff
@@ -3,243 +3,259 @@
...
-      ValueWhenMaximumOrMinimum,            !- Aggregation Type for Variable or Meter 12
-      Cogeneration:DistrictHeating,!- Variable or Meter 13 Name
-      ValueWhenMaximumOrMinimum;            !- Aggregation Type for Variable or Meter 13)idf",
+Output:Table:Monthly,
+  Building Energy Performance - District Heating Water Peak Demand, !- Name
+  2,                                      !- Digits After Decimal
+  DistrictHeatingWater:Facility,          !- Variable or Meter Name 1
+  Maximum,                                !- Aggregation Type for Variable or Meter 1
+  InteriorLights:DistrictHeatingWater,    !- Variable or Meter Name 2
+  ValueWhenMaximumOrMinimum,              !- Aggregation Type for Variable or Meter 2
```
Comment (jmarrec) [inline src/workflow/RunPreProcessMonthlyReports.hpp:188]: Same meter rename for the peak one

File: src/workflow/RunPreProcessMonthlyReports.hpp [hunk 6]
```diff
@@ -3,243 +3,259 @@
...
-  //These are needed for the calibration report
+  // These are needed for the calibration report
   "Output:Meter:MeterFileOnly,NaturalGas:Facility,Daily;",
   "Output:Meter:MeterFileOnly,Electricity:Facility,Timestep;",
   "Output:Meter:MeterFileOnly,Electricity:Facility,Daily;",
 
   // Always add in the timestep facility meters
   "Output:Meter,Electricity:Facility,Timestep;",
   "Output:Meter,NaturalGas:Facility,Timestep;",
+  "Output:Meter,DistrictHeatingWater:Facility,Timestep;",
```
Comment (jmarrec) [inline src/workflow/RunPreProcessMonthlyReports.hpp:257]: And it's needed in meters forced too.

File: src/workflow/RunPreProcessMonthlyReports.hpp (hunk 2 above)
Comment (DavidGoldwasser) [reply src/workflow/RunPreProcessMonthlyReports.hpp:67]: Added @joseph-robertson on to review of this PR as well

**Review by joseph-robertson — APPROVED** [review]
New script for generating `RunPreprocessMonthlyREports.hpp` 👍 
New workflow tests for preventing future breaks 👍 
Add missing `Refrigeration:Electricity` to Electricity Peak Demand 👍 
Leaving it up to @DavidGoldwasser to comment on use of `OpenStudio::EndUseFuelType::valueDescription` in reporting measures.

**Pull request merged**